src/NLog/Config/LoggingConfigurationLoaderFileWatcher.cs

```diff
@@ -0,0 +1,296 @@
+﻿// 
+// Copyright (c) 2004-2018 Jaroslaw Kowalski <jaak@jkowalski.net>, Kim Christensen, Julian Verdurmen
+// 
+// All rights reserved.
+// 
+// Redistribution and use in source and binary forms, with or without 
+// modification, are permitted provided that the following conditions 
+// are met:
+// 
+// * Redistributions of source code must retain the above copyright notice, 
+//   this list of conditions and the following disclaimer. 
+// 
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+#if !SILVERLIGHT && !__IOS__ && !__ANDROID__ && !NETSTANDARD1_3
+
+namespace NLog.Config
+{
+    using System;
+    using System.Linq;
+    using System.Threading;
+    using NLog.Common;
+    using NLog.Internal;
+
+    /// <summary>
+    /// Enables FileWatcher for the currently loaded NLog Configuration File,
+    /// and supports automatic reload on file modification.
+    /// </summary>
+    internal class LoggingConfigurationLoaderFileWatcher : LoggingConfigurationLoaderFile
```

Comment:
I'm sorry, but I think these class names are confusing.

It's from the names not clear those two classes are ConfigurationLoaders, and that this class is a specilized version of the "normal " file loader.

 I know those are internal, but will also popup in stacktraces and we need to maintain them ;)

proposal, but other proposals also accepted:

- LoggingConfigurationFileLoader
- LoggingConfigurationWatchableFileLoader

Maybe another option is to create "sub" namespaces to help in the naming.

Reply:
Made one more rename attempt:

- LoggingConfigurationFileLoader
- LoggingConfigurationFileWatchingLoader

Reply:
Renamed classes as proposed.